sam/models/base_model.py

```diff
@@ -486,6 +487,11 @@ def get_feature_names_out(self, input_features=None) -> List[str]:
         Function for obtaining feature names. Generally used instead of the attribute, and more
         compatible with the sklearn API.
 
+        It is assumed that in case of Pipelines for feature engineer, the last
+        ColumnTransformer step contains the feature names. If any element from
+        the Pipeline is an instance of ColumnTransformer, the last step is used for
```

Comment:
Add to doc string that if no ColumnTransformer is found the last step that contains the `get_feature_names_out` attribute is used.

Reply:
Added